src/components/forms/NumberField/NumberField.js

```diff
@@ -62,15 +56,18 @@ const NumberField = ({
     invalid,
     min,
     step,
-    type: 'text',
+    type: useNumberType ? 'number' : 'text',
 
     // These are for the NumericFormat component
     customInput: Textbox,
-    isAllowed,
-    allowNegative,
-    decimalSeparator,
-    thousandSeparator,
-    decimalScale: isAllowedToTypeDecimals ? 2 : 0,
+    // only pass the localized separators when we're not using type="number" input
+    ...(!useNumberType
+      ? {
+          decimalSeparator,
+          thousandSeparator,
+          decimalScale: isInteger ? undefined : 2,
+        }
+      : {}),
```

Comment:
I noticed I couldn't enter `1000` when using a number input, the localized formatting was probably breaking stuff, so the idea is to only leverage that functionality when we're using a text-based input component.